### ISSUE TITLE
/dist/main.js not being built properly, expanded src also being served

### ISSUE THREAD

**Issue by humphd** [mozilla/thimble.webmaker.org#1035]
In production, we're supposed to load just one JS file for our require.js modules:

https://github.com/mozilla/thimble.webmaker.org/blob/bramble/routes/main/editor.js#L65

We have confirmed that we are building that file, and that we have the env set correctly.  However, we are still triggering a download of the other expanded src files.

**Comment by humphd** [comment]
@jrburke, I'm running up against my own ignorance trying to get our Thimble app's requirejs build working.  If you have a moment, I could use a pointer on how to fix things--I'm sure it's me doing things incorrectly.

I have a build step:

https://github.com/mozilla/thimble.webmaker.org/blob/bramble/Gruntfile.js#L33-L67

And it's building `main.js`:

https://github.com/mozilla/thimble.webmaker.org/blob/bramble/public/editor/scripts/main.js

I am doing a two-level `require` call, because I need to deal with some browsers not being able to load bits of what I need first, and only then do I want to try and load the modules.

The problem is that the resulting `dist/main.js` contains only the two modules mentioned in the outer require call, and the inner ones don't get included (the server requests them from the server's src tree directly).

I saw what you suggested in https://groups.google.com/forum/#!topic/requirejs/ziKi6qWc9vw and tried passing `"require"` as the first module.  But that didn't do it.

**Comment by jrburke** [comment]
By default, nested require([]) calls are not scanned for concatenation as they are assumed to be dynamic things that should be loaded on demand. Two options:

1) set [findNestedDependencies: true](https://github.com/jrburke/r.js/blob/2fdfddd959f371b775027c0145b00aed334a58f6/build/example.build.js#L339) in the build config. However, if you do have other nested require([]) calls in the files that will be part of that main.js layer that you really do want to keep dynamically loaded, then I suggest option 2).

2) Manually list them in the build layer via [include](). So, it would be something like:

```javascript
          modules: [{
            name: "../../main",
            include: ["bramble-editor", "project", "sso-override", "fc/project-rename"]
          }],
```

Of course the hazard with that is keeping the build file in sync with changes you do to that inner require([]) call in main.js.

**Comment by gideonthomas** [comment]
Fixed in abb627e0a29c3776f900134384858eb354db1fb7

**Comment by humphd** [comment]
@jrburke, as usual, you're a hero.  Thanks wasting your time on this, it's really appreciated!
